app/test_main.py

```diff
@@ -2,0 +1,39 @@
+from unittest.mock import patch
+
+from app.main import can_access_google_page
+
+
+def test_can_access_google_page_valid_url_and_internet() -> None:
+    with patch("app.main.valid_google_url",
+               return_value=True) as mock_valid_url, \
+         patch("app.main.has_internet_connection",
+               return_value=True) as mock_internet:
+
+        result = can_access_google_page("https://www.google.com")
+
+        mock_valid_url.assert_called_once_with("https://www.google.com")
+        mock_internet.assert_called_once()
+        assert result == "Accessible"
+
+
+def test_can_access_google_page_invalid_url() -> None:
```

Comment:
The test name `test_can_access_google_page_invalid_url` could be more descriptive. Consider renaming it to something like `test_not_accessible_with_invalid_url` to better convey the scenario being tested.